ethcore/src/miner/miner.rs

```diff
@@ -482,6 +489,7 @@ impl Miner {
 				// Save proposal for later seal submission and broadcast it.
 				Seal::Proposal(seal) => {
 					trace!(target: "miner", "Received a Proposal seal.");
+					*self.next_mandatory_reseal.write() = Instant::now() + self.options.reseal_max_period;
```

Comment:
PoW blocks awaiting work are not sealed. So PoW only uses the `Seal::Regular` code path.
Both PoW blocks and proposal blocks go into the "work" queue to be later released with a final seal.